sdk/samples/iot/iot_sample_foundation.c

```diff
@@ -0,0 +1,450 @@
+        AZ_RETURN_IF_FAILED(az_span_atou32(duration, &(env_vars->sas_key_duration_minutes)));
+        break;
+
+      default:
+        LOG_ERROR("Provisioning sample name undefined.");
+        return AZ_ERROR_ARG;
+    }
+  }
+  else
+  {
+    LOG_ERROR("Sample type undefined.");
+    return AZ_ERROR_ARG;
+  }
+
+  env_vars->x509_trust_pem_file_path = AZ_SPAN_FROM_BUFFER(x509_trust_pem_file_path_buffer);
+  AZ_RETURN_IF_FAILED(read_configuration_entry(
+      ENV_IOT_DEVICE_X509_TRUST_PEM_FILE_PATH, "", false, &(env_vars->x509_trust_pem_file_path)));
+
+  LOG(" "); // Log formatting
+  return AZ_OK;
+}
+
+az_result read_configuration_entry(
+    const char* env_name,
+    char* default_value,
+    bool hide_value,
+    az_span* out_value)
+{
+  PRECONDITION_NOT_NULL(env_name);
+  PRECONDITION_NOT_NULL(out_value);
+
+  char* env_value = getenv(env_name);
+
+  if (env_value == NULL && default_value != NULL)
+  {
+    env_value = default_value;
+  }
+
+  if (env_value != NULL)
+  {
+    (void)printf("%s = %s\n", env_name, hide_value ? "***" : env_value);
+    az_span env_span = az_span_create_from_str(env_value);
+
+    // assumes *out_value is a valid span
+    AZ_RETURN_IF_NOT_ENOUGH_SIZE(*out_value, az_span_size(env_span));
+    az_span_copy(*out_value, env_span);
+    *out_value = az_span_slice(
+        *out_value, 0, az_span_size(env_span)); // Trim any excess. Important for numbers
+  }
+  else
+  {
+    LOG_ERROR("(missing) Please set the %s environment variable.", env_name);
+    return AZ_ERROR_ARG;
+  }
+
+  return AZ_OK;
+}
+
+az_result create_mqtt_endpoint(
+    sample_type type,
+    const sample_environment_variables* env_vars,
+    char* endpoint,
+    size_t endpoint_size)
+{
+  PRECONDITION_NOT_NULL(env_vars);
+  PRECONDITION_NOT_NULL(endpoint);
+
+  if (type == PAHO_IOT_HUB)
+  {
+    int32_t required_size = az_span_size(mqtt_url_prefix) + az_span_size(env_vars->hub_hostname)
+        + az_span_size(mqtt_url_suffix) + (int32_t)sizeof('\0');
+
+    if (required_size > (int32_t)endpoint_size)
+    {
+      return AZ_ERROR_INSUFFICIENT_SPAN_SIZE;
+    }
+
+    az_span hub_mqtt_endpoint = az_span_create((uint8_t*)endpoint, (int32_t)endpoint_size);
+    az_span remainder = az_span_copy(hub_mqtt_endpoint, mqtt_url_prefix);
+    remainder = az_span_copy(remainder, env_vars->hub_hostname);
+    remainder = az_span_copy(remainder, mqtt_url_suffix);
+    az_span_copy_u8(remainder, '\0');
+  }
+  else if (type == PAHO_IOT_PROVISIONING)
+  {
+    int32_t required_size = az_span_size(provisioning_global_endpoint) + (int32_t)sizeof('\0');
+
+    if (required_size > (int32_t)endpoint_size)
+    {
+      return AZ_ERROR_INSUFFICIENT_SPAN_SIZE;
+    }
+
+    az_span provisioning_mqtt_endpoint = az_span_create((uint8_t*)endpoint, (int32_t)endpoint_size);
+    az_span remainder = az_span_copy(provisioning_mqtt_endpoint, provisioning_global_endpoint);
+    az_span_copy_u8(remainder, '\0');
+    ;

```

Comment:
hmm, i'm not sure how that's in there because a build attempt would have caught it.  I think you may be looking at an older version?  221 doesn't match up with what I have.

Reply:
a blank semi-colon is technically valid so the build wouldn't catch it. It's currently in master with the semicolon:
https://github.com/Azure/azure-sdk-for-c/blob/4c9acf3a9c3deb89f82286f389507ce4790241be/sdk/samples/iot/iot_sample_foundation.c#L221